src/Hl7.Fhir.Specification/Specification/Snapshot/SnapshotGeneratorSettings.cs

```diff
@@ -81,6 +82,13 @@ public void CopyTo(SnapshotGeneratorSettings other)
         /// <summary>Enable this setting to automatically generate missing element id values.</summary>
         public bool GenerateElementIds { get; set; } = true;
 
+        /// <summary>
+        /// Indicates the intended use of the snapshot generator output: snapshot or differential.
+        /// Snapshot generator business logic may vary depending on the intended use.
+        /// The default value is set to a backwards compatible setting (i.e. snapshot and differential).
+        /// </summary>
+        public SnapshotIntendedUse IntendedUse { get; set; } = SnapshotIntendedUse.BackwardsCompatible;
```

Comment:
This requires a lot of context to understand (I did not read the discussions leading up to this). I guess that if I don't get it, no one will (except for you two ;-)).  What are the differences? Why would a snapshot generator touch the differential? (just the first questions that came up).  Is there a way to make this setting a bit more explict? Something like  naming it `FixDiscrepancyBetweenXAndIButYouShouldntWorryAboutThis`?  Is this really about intended use?

Ok, maybe I should now read the discussion leading up to this ;-)

Reply:
There is a specific bug in R4 that needs fixing but we tried to make the solution as generic as possible. Any feedback is of course welcome!

Reply:
It's just that this is part of the public interface, and there's no way to understand what it does based on the current public xmldoc comment.  Maybe expand on it by documenting the `enum` ?

Reply:
How about:

```
    /// <summary>
    /// Indicates the intended use of the snapshot generator output.
    /// Snapshot generator business logic may vary depending on the intended use.
    /// For example: if an "element.constraint.source" references the resource itself,
    /// then the source value should be set for the snapshot but not for the differential.
    /// </summary>
    public enum SnapshotIntendedUse
    {
        /// <summary>
        /// Indicates that the snapshot generator output is intended to be used for the snapshot component of a structure definition.
        /// </summary>
        Snapshot = 1 << 0,
        /// <summary>
        /// Indicates that the snapshot generator output is intended to be used for the differential component of a structure definition.
        /// </summary>
        Differential = 1 << 1,
        /// <summary>
        /// This is the (default) backwards compatibility setting.
        /// </summary>
        BackwardsCompatible = Snapshot | Differential
    }
```

Reply:
@ewoutkramer I can imagine that you don't like this solution but then we would have to restart the discussion with Grahame (https://chat.fhir.org/#narrow/stream/179177-conformance/topic/STU3.20Qustionnaire.20snapshot.20generation/near/279683046)

Reply:
From what I understand, is that Forge uses the result of the SnapshotGenerator to fill missing (and incorrent?) elements in the differential. It then behaves differently then it would when it creating a snapshot.

Reply:
Maybe that is better. I am careful about adding properties to the settings (they only tend to grow in sieze) that are of no use to the general public.  A carefully documented member (maybe even documented as "internal use only") may be better.

Reply:
@ewoutkramer @mmsmits To correct for this in Forge is not hard to do either. Who is responsible for making this correction: Forge or the SDK?

Reply:
The discussion on Zulip seems to be about snapshotting behaviour that others outside of Forge should be able to do, I think it is the responsibility of the SDK. But I might be wrong, I don't see the connection with this issue?